commit a2b7cd8457906c732fe6b1f6d2968eeaa58c02a6
Author: Matthew Fernandez <matthew.fernandez@gmail.com>
Date:   Mon Apr 27 17:02:34 2026 +1000

    tests: Drop casts around `XML_GetUserData` that C99 does not need
    
    The `XML_GetUserData` macro returns a `void *`. Under C99, `void *`
    implicitly coerces to all other pointer types.
    
    This is a follow-up to changes merged in
    89e6f354f44f27eda61e1b94a473f88b2f9a8bc6.

diff --git a/expat/tests/basic_tests.c b/expat/tests/basic_tests.c
index a4a19472..5fd55933 100644
--- a/expat/tests/basic_tests.c
+++ b/expat/tests/basic_tests.c
@@ -3435,8 +3435,7 @@ external_bom_checker(XML_Parser parser, const XML_Char *context,
     fail("Could not create external entity parser");
 
   if (! xcstrcmp(systemId, XCS("004-2.ent"))) {
-    struct bom_testdata *const testdata
-        = (struct bom_testdata *)XML_GetUserData(parser);
+    struct bom_testdata *const testdata = XML_GetUserData(parser);
     const char *const external = testdata->external;
     const int split = testdata->split;
     testdata->nested_callback_happened = XML_TRUE;
diff --git a/expat/tests/handlers.c b/expat/tests/handlers.c
index e456df21..2701c695 100644
--- a/expat/tests/handlers.c
+++ b/expat/tests/handlers.c
@@ -423,7 +423,7 @@ int XMLCALL
 external_entity_optioner(XML_Parser parser, const XML_Char *context,
                          const XML_Char *base, const XML_Char *systemId,
                          const XML_Char *publicId) {
-  ExtOption *options = (ExtOption *)XML_GetUserData(parser);
+  ExtOption *options = XML_GetUserData(parser);
   XML_Parser ext_parser;
 
   UNUSED_P(base);
@@ -449,7 +449,7 @@ int XMLCALL
 external_entity_loader(XML_Parser parser, const XML_Char *context,
                        const XML_Char *base, const XML_Char *systemId,
                        const XML_Char *publicId) {
-  ExtTest *test_data = (ExtTest *)XML_GetUserData(parser);
+  ExtTest *test_data = XML_GetUserData(parser);
   XML_Parser extparser;
 
   UNUSED_P(base);
@@ -477,7 +477,7 @@ external_entity_faulter(XML_Parser parser, const XML_Char *context,
                         const XML_Char *base, const XML_Char *systemId,
                         const XML_Char *publicId) {
   XML_Parser ext_parser;
-  ExtFaults *fault = (ExtFaults *)XML_GetUserData(parser);
+  ExtFaults *fault = XML_GetUserData(parser);
 
   UNUSED_P(base);
   UNUSED_P(systemId);
@@ -650,7 +650,7 @@ external_entity_suspending_faulter(XML_Parser parser, const XML_Char *context,
                                    const XML_Char *systemId,
                                    const XML_Char *publicId) {
   XML_Parser ext_parser;
-  ExtFaults *fault = (ExtFaults *)XML_GetUserData(parser);
+  ExtFaults *fault = XML_GetUserData(parser);
   void *buffer;
   int parse_len = (int)strlen(fault->parse_text);
 
@@ -981,7 +981,7 @@ external_entity_valuer(XML_Parser parser, const XML_Char *context,
         == XML_STATUS_ERROR)
       xml_failure(ext_parser);
   } else if (! xcstrcmp(systemId, XCS("004-2.ent"))) {
-    ExtFaults *fault = (ExtFaults *)XML_GetUserData(parser);
+    ExtFaults *fault = XML_GetUserData(parser);
     enum XML_Status status;
     enum XML_Error error;
 
@@ -1083,7 +1083,7 @@ int XMLCALL
 external_entity_public(XML_Parser parser, const XML_Char *context,
                        const XML_Char *base, const XML_Char *systemId,
                        const XML_Char *publicId) {
-  const char *text1 = (const char *)XML_GetUserData(parser);
+  const char *text1 = XML_GetUserData(parser);
   const char *text2 = "<!ATTLIST doc a CDATA 'value'>";
   const char *text = NULL;
   XML_Parser ext_parser;
@@ -1139,7 +1139,7 @@ int XMLCALL
 external_entity_oneshot_loader(XML_Parser parser, const XML_Char *context,
                                const XML_Char *base, const XML_Char *systemId,
                                const XML_Char *publicId) {
-  ExtHdlrData *test_data = (ExtHdlrData *)XML_GetUserData(parser);
+  ExtHdlrData *test_data = XML_GetUserData(parser);
   XML_Parser ext_parser;
 
   UNUSED_P(base);
@@ -1164,7 +1164,7 @@ int XMLCALL
 external_entity_loader2(XML_Parser parser, const XML_Char *context,
                         const XML_Char *base, const XML_Char *systemId,
                         const XML_Char *publicId) {
-  ExtTest2 *test_data = (ExtTest2 *)XML_GetUserData(parser);
+  ExtTest2 *test_data = XML_GetUserData(parser);
   XML_Parser extparser;
 
   UNUSED_P(base);
@@ -1191,7 +1191,7 @@ int XMLCALL
 external_entity_faulter2(XML_Parser parser, const XML_Char *context,
                          const XML_Char *base, const XML_Char *systemId,
                          const XML_Char *publicId) {
-  ExtFaults2 *test_data = (ExtFaults2 *)XML_GetUserData(parser);
+  ExtFaults2 *test_data = XML_GetUserData(parser);
   XML_Parser extparser;
 
   UNUSED_P(base);
@@ -1309,7 +1309,7 @@ int XMLCALL
 external_entity_dbl_handler(XML_Parser parser, const XML_Char *context,
                             const XML_Char *base, const XML_Char *systemId,
                             const XML_Char *publicId) {
-  int *pcallno = (int *)XML_GetUserData(parser);
+  int *pcallno = XML_GetUserData(parser);
   int callno = *pcallno;
   const char *text;
   XML_Parser new_parser = NULL;
@@ -1366,7 +1366,7 @@ int XMLCALL
 external_entity_dbl_handler_2(XML_Parser parser, const XML_Char *context,
                               const XML_Char *base, const XML_Char *systemId,
                               const XML_Char *publicId) {
-  int *pcallno = (int *)XML_GetUserData(parser);
+  int *pcallno = XML_GetUserData(parser);
   int callno = *pcallno;
   const char *text;
   XML_Parser new_parser;
@@ -1461,7 +1461,7 @@ int XMLCALL
 external_entity_alloc(XML_Parser parser, const XML_Char *context,
                       const XML_Char *base, const XML_Char *systemId,
                       const XML_Char *publicId) {
-  const char *text = (const char *)XML_GetUserData(parser);
+  const char *text = XML_GetUserData(parser);
   XML_Parser ext_parser;
   int parse_res;
 
@@ -1516,8 +1516,7 @@ accounting_external_entity_ref_handler(XML_Parser parser,
   UNUSED_P(base);
   UNUSED_P(publicId);
 
-  const struct AccountingTestCase *const testCase
-      = (const struct AccountingTestCase *)XML_GetUserData(parser);
+  const struct AccountingTestCase *const testCase = XML_GetUserData(parser);
 
   const char *externalText = NULL;
   if (xcstrcmp(systemId, XCS("first.ent")) == 0) {
